File: tic-tac-toe-core/src/main/java/by/xxx/pupil/ai/minimax/impl/AllAvailableMovesFinder.java
package by.xxx.pupil.ai.minimax.impl;

import by.xxx.pupil.BoardUtils;
import by.xxx.pupil.ai.minimax.MovesFinder;
import by.xxx.pupil.model.Board;
import by.xxx.pupil.model.Move;

import java.util.ArrayList;
import java.util.List;

public class AllAvailableMovesFinder implements MovesFinder {

    @Override
    public List<Move> getMoves(Board board, boolean isPerson) {
        List<Move> availableMoves = new ArrayList<>();
        for (int i = 0; i < board.getHeight(); i++) {
            for (int j = 0; j < board.getWidth(); j++) {
                if (BoardUtils.isCellEmpty(board, i, j)) {
                    availableMoves.add(new Move(i, j, isPerson));
                }
            }
        }

        return availableMoves;
    }
}

File: tic-tac-toe-core/src/main/kotlin/by/pupil/ai/minimax/findmoves/AllAvailableMovesFinder.kt
package by.pupil.ai.minimax.findmoves

import by.pupil.model.Board
import by.pupil.model.Move
import by.pupil.model.Player
import by.pupil.model.isCellEmpty

class AllAvailableMovesFinder : MovesFinder {
    override fun getMoves(
        board: Board,
        player: Player,
    ): List<Move> {
        val availableMoves: MutableList<Move> = ArrayList()
        for (i in 0 until board.height) {
            for (j in 0 until board.width) {
                if (isCellEmpty(board, i, j)) {
                    availableMoves.add(Move(i, j, player))
                }
            }
        }
        return availableMoves
    }
}
